### PR TITLE
NMS-12679: Update database and send events only on changes

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-12679

### CODE REVIEW

File: features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java [hunk 1]
```diff
@@ -261,12 +261,22 @@ public LocationAwarePollerClient getLocationAwarePollerClient() {
...
         final OnmsMonitoringLocation location = this.monitoringLocationDao.get(locationName);
 
-        final OnmsLocationSpecificStatus status = new OnmsLocationSpecificStatus();
-        status.setLocation(location);
-        status.setMonitoredService(polledService.getMonSvc());
-        status.setPollResult(pollResult);
+        final OnmsLocationSpecificStatus oldLocationSpecificStatus = this.locationSpecificStatusDao.getMostRecentStatusChange(location, polledService.getMonSvc());
 
-        this.locationSpecificStatusDao.saveStatusChange(status);
+        if (oldLocationSpecificStatus == null || oldLocationSpecificStatus.getPollResult().getStatusCode() != pollResult.getStatusCode()) {
```
Comment (fooker) [inline features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java:266]: Should we update if the reason changes?

File: features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java [hunk 2]
```diff
@@ -261,12 +261,22 @@ public LocationAwarePollerClient getLocationAwarePollerClient() {
     protected void reportResult(final String locationName, final RemotePolledService polledService, final PollStatus pollResult) {
         final OnmsMonitoringLocation location = this.monitoringLocationDao.get(locationName);
 
-        final OnmsLocationSpecificStatus status = new OnmsLocationSpecificStatus();
-        status.setLocation(location);
-        status.setMonitoredService(polledService.getMonSvc());
-        status.setPollResult(pollResult);
+        final OnmsLocationSpecificStatus oldLocationSpecificStatus = this.locationSpecificStatusDao.getMostRecentStatusChange(location, polledService.getMonSvc());
```
Comment (fooker) [inline features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java:264]: The whole select/update cycle should run in a single transaction

File: opennms-base-assembly/src/main/filtered/etc/create.sql [hunk 3]
```diff
@@ -343,7 +343,7 @@ CREATE TABLE scanreports (
     CONSTRAINT scanreports_monitoringlocations_fkey FOREIGN KEY (location) REFERENCES monitoringlocations (id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
-CREATE UNIQUE INDEX scanreports_id_idx on scanreport(id);
+CREATE UNIQUE INDEX scanreports_id_idx on scanreports(id);
```
Comment (fooker) [inline opennms-base-assembly/src/main/filtered/etc/create.sql:346]: Not sure if...

File: features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java (hunk 1 above)
Comment (christianpape) [reply features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java:266]: done.

File: features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java (hunk 2 above)
Comment (christianpape) [reply features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java]: done.

File: opennms-base-assembly/src/main/filtered/etc/create.sql (hunk 3 above)
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/create.sql:346]: Oh, come on, it's not really hurting anyone.

**Pull request merged**